aptos-move/move-examples/token_objects/sources/royalty.move

```diff
@@ -0,0 +1,102 @@
+/// This defines an object-based Royalty. The royalty can be applied to either a collection or a
+/// token. Applications should read the royalty from the token, as it will read the appropriate
+/// royalty.
+module token_objects::royalty {
+    use std::option::{Self, Option};
+
+    use aptos_framework::object::{Self, ConstructorRef, ExtendRef, Object};
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The royalty of a token within this collection -- this optional
+    struct Royalty has copy, drop, key {
+        numerator: u64,
+        denominator: u64,
+        /// The recipient of royalty payments. See the `shared_account` for how to handle multiple
+        /// creators.
+        payee_address: address,
+    }
+
+    struct MutatorRef has drop, store {
+        inner: ExtendRef,
+    }
+
+    /// Add a royalty, given a ConstructorRef.
+    public fun init(ref: &ConstructorRef, royalty: Royalty) {
+        let signer = object::generate_signer(ref);
+        move_to(&signer, royalty);
+    }
+
+    /// Set the royalty if it does not exist, replace it otherwise.
+    public fun update(mutator_ref: &MutatorRef, royalty: Royalty) acquires Royalty {
+        let addr = object::address_from_extend_ref(&mutator_ref.inner);
+        if (exists<Royalty>(addr)) {
+            move_from<Royalty>(addr);
+        };
+
+        let signer = object::generate_signer_for_extending(&mutator_ref.inner);
+        move_to(&signer, royalty);
+    }
+
+    public fun create(numerator: u64, denominator: u64, payee_address: address): Royalty {
+        Royalty { numerator, denominator, payee_address }
+    }
```

Comment:
Can we ensure numerator <= denominator?  and the denominator isn't 0?

Reply:
done